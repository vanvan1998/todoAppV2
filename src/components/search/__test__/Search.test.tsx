import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Search } from '..';

describe('Test Search component', () => {
  test.only('should match its snapshot', async () => {
    const component = render(<Search searchString='searchString' handleSearch={jest.fn()} />);
    const input = screen.getByLabelText('Search');
    fireEvent.change(input, { target: { value: '123' } });

    expect(component).toMatchSnapshot();
  });
});
