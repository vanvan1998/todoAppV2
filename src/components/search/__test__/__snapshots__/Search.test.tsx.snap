// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="sc-gLDymS hwUfDl"
        >
          <svg
            fill="currentColor"
            height="18"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M456.69 421.39 362.6 327.3a173.81 173.81 0 0 0 34.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 0 0 327.3 362.6l94.09 94.09a25 25 0 0 0 35.3-35.3zM97.92 222.72a124.8 124.8 0 1 1 124.8 124.8 124.95 124.95 0 0 1-124.8-124.8z"
            />
          </svg>
          <input
            aria-label="Search"
            class="sc-iAEzLF gjdHlk"
            placeholder="Search..."
            type="text"
            value="searchString"
          />
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="sc-gLDymS hwUfDl"
      >
        <svg
          fill="currentColor"
          height="18"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M456.69 421.39 362.6 327.3a173.81 173.81 0 0 0 34.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 0 0 327.3 362.6l94.09 94.09a25 25 0 0 0 35.3-35.3zM97.92 222.72a124.8 124.8 0 1 1 124.8 124.8 124.95 124.95 0 0 1-124.8-124.8z"
          />
        </svg>
        <input
          aria-label="Search"
          class="sc-iAEzLF gjdHlk"
          placeholder="Search..."
          type="text"
          value="searchString"
        />
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
